File: UltimateFestivalOrganizer/UltimateFestivalOrganizer.Web/src/main/java/at/wlumetsberger/ufo/services/interfaces/IQueryService.java
package at.wlumetsberger.ufo.services.interfaces;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.List;

import at.wlumetsberger.ufo.models.Artist;
import at.wlumetsberger.ufo.models.Catagory;
import at.wlumetsberger.ufo.models.Performance;
import at.wlumetsberger.ufo.models.Venue;
import at.wlumetsberger.ufo.services.exceptions.ServiceConnectionException;

public interface IQueryService extends Serializable{

	List<Artist> queryArtists() throws ServiceConnectionException;
	List<Catagory> queryCatagories() throws ServiceConnectionException;
	List<Venue> queryVenues() throws ServiceConnectionException;
	List<Performance> queryPerformancesByDay(LocalDate day) throws ServiceConnectionException;
	List<Performance> queryPerformancesByArtist(int artistId) throws ServiceConnectionException;
	List<Performance> queryPerformancesByVenue(int venueId) throws ServiceConnectionException;
	List<Performance> queryPerformancesByCatagory(int catagoryId) throws ServiceConnectionException;	
}

File: UltimateFestivalOrganizer/UltimateFestivalOrganizer.BusinessLogik/IQueryService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UltimateFestivalOrganizer.DAL.Common.Domain;

namespace UltimateFestivalOrganizer.BusinessLogik
{
    public interface IQueryService
    {
        IList<Artist> QueryArtists();
        Artist QueryArtistById(string id);

        IList<Performance> QueryPerformances();
        IList<Performance> QueryPerfomancesByDay(DateTime day);
        IList<Performance> QueryPerfomancesByArtist(int artistId);
        IList<Performance> QueryPerfomancesByVenue(int venueId);
        IList<Performance> QueryPerfomancesByCatagory(int catagoryId);

        IList<Catagory> QueryCatagories();
        IList<Venue> QueryVenues();
        bool CheckPostponeIsPossible(int performanceId, DateTime date, int venueId);
        bool PostponePerformance(int performanceId, DateTime date, int venueId);
        bool CanclePerformance(int performanceId);
    }
}
